src/lib/analytics.js

```diff
@@ -104,14 +103,24 @@ document.addEventListener(
   },
 );
 
-// Update Analytics dimension if signed-in state changes. This doesn't cause a
-// new pageview implicitly but annotates all further events.
-// We log pageviews only in bootstrap.js (on entry, for all browsers) and in
-// loader.js (for dynamic SPA page loads, part of our core bundle).
-store.subscribe(({isSignedIn}) => {
-  // nb. Analytics requires dimension values to be strings.
-  ga('set', dimensions.SIGNED_IN, isSignedIn ? '1' : '0');
-});
+/**
+ * Add a listener to detect back/forward cache restores and track them
+ * as pageviews with the "bfcache" navigation type set (in case we need
+ * to distinguish them from regular pageviews).
+ * https://web.dev/bfcache/#how-bfcache-affects-analytics-and-performance-measurement
+ */
+window.addEventListener(
+  'pageshow',
+  /**
+   * @param {PageTransitionEvent} e
+   */
+  (e) => {
+    if (e.persisted) {
+      ga('set', dimensions.NAVIGATION_TYPE, 'bfcache');
```

Comment:
THis will break the continuity of our analytics for the signed_in dimention, right?

Reply:
I re-added the old code since we're planning on keeping the sign-in button on pages that currently have it.

I did want to point out, however, that the existing code isn't actually very useful. It tracks the state change when a user signs in or out, but it doesn't set the stored state at page load time (for the initial pageview), so if a user doesn't sign in or out during their session, we don't actually know whether they were signed in. For example, only 17 out of the 600K pageviews received in the last 7 days have any sign-in information.

If we're planning to use the sign-in functionality more in the future, we should come back to this issue (which would probably require a refactor of the existing analytics code).